File: 206/206.js
var start = Math.floor(Math.sqrt(1020304050607080900));
var stop  = Math.floor(Math.sqrt(1929394959697989999));

for(var i = start; i < stop; i+=100) {
	var temp = ('' + (i * i)).split('');
	for(var j = 1; j < temp.length; j++) {
		temp.splice(j,1);
	}
	if(temp.join('') == '1234567890') {
		console.log(i);
		break;
	}
}

File: 206/206.rb
# Straight forward brute force 
start = (1020304050607080900**0.5).floor
stop  = (1929394959697989999**0.5).floor
(start..stop).step(10) do |i|
	temp = (i * i).to_s
	found = true
	9.times do |j|
		if temp[2*j..2*j].to_i != j+1
			found = false
			break
		end
	end
	if found
		puts i
		break
	end
end
